src/bun.js/web_worker.zig

```diff
@@ -303,30 +276,29 @@ pub const WebWorker = struct {
         // always doing a first tick so we call CppTask without delay after dispatchOnline
         vm.tick();
 
-        {
-            while (true) {
-                while (vm.eventLoop().tasks.count > 0 or vm.active_tasks > 0 or vm.uws_event_loop.?.active > 0) {
-                    vm.tick();
-
-                    vm.eventLoop().autoTickActive();
-                }
-
-                if (!this.allowed_to_exit) {
-                    this.flushLogs();
-                    vm.eventLoop().tickPossiblyForever();
-                    continue;
+        while (true) {
+            while (vm.eventLoop().tasks.count > 0 or vm.active_tasks > 0 or vm.uws_event_loop.?.active > 0) {
+                this.eventloop_poll_ref.refConcurrently(this.parent);
```

Comment:
this should not be here

Reply:
i meant to have this at the end of the while(true)

Reply:
It should not call .refConcurrently() on every tick of the event loop? It should only need to call it once while there are pending tasks.